visualization/app/codeCharta/codeCharta.service.ts

```diff
@@ -52,36 +76,38 @@ export class CodeChartaService {
 		}
 	}
 
-	private getAttributeTypes(attributeTypes: AttributeTypes): AttributeTypes {
-		const newAttributeTypes: any = {}
-
-		if (_.isEmpty(attributeTypes) || !attributeTypes) {
+	private getAttributeTypes(attributeTypes: AttributeTypes | OldAttributeTypes): AttributeTypes {
+		if (_.isEmpty(attributeTypes) || !attributeTypes || Array.isArray(attributeTypes.nodes) || Array.isArray(attributeTypes.edges)) {
 			return {
 				nodes: {},
 				edges: {}
 			}
-		} else {
-			if (!attributeTypes.nodes) {
-				newAttributeTypes.nodes = {}
-			} else {
-				newAttributeTypes.nodes = attributeTypes.nodes
-			}
+		}
 
-			if (!attributeTypes.edges) {
-				newAttributeTypes.edges = {}
-			} else {
-				newAttributeTypes.edges = attributeTypes.edges
-			}
+		return {
+			nodes: !attributeTypes.nodes ? {} : attributeTypes.nodes,
+			edges: !attributeTypes.edges ? {} : attributeTypes.edges
 		}
-		return newAttributeTypes
 	}
 
 	private potentiallyUpdateBlacklistTypes(blacklist): BlacklistItem[] {
 		blacklist.forEach(x => {
-			if (x.type === "hide") {
+			if (x.type === ExportBlacklistType.hide) {
 				x.type = BlacklistType.flatten
 			}
 		})
 		return blacklist
 	}
+
+	private setDefaultScenario() {
+		const metricData = this.metricService.getMetricData()
+		const { areaMetric, heightMetric, colorMetric } = ScenarioHelper.getDefaultScenarioSetting().dynamicSettings
+		const names = [areaMetric, heightMetric, colorMetric]
+
+		const existsInMetricData = (metric: string) => metricData.map(x => x.name).includes(metric)
```

Comment:
The map increases the work significantly, since we do not only have to do the lookup but also have to go through all entries each time. It's therefore `O(n * m * 2)` and the map operation is more expensive then the includes due to allocating a new array and transforming the content. There are not a lot of entries, it won't hurt much due to that but it's a good practice not to code like that in general.

```suggestion
		const metricNames = new Set(metricData.map(x => x.name))
		const existsInMetricData = (metricName: string) => metricNames.has(metricName)
```